Flows by port: add condition to check switch ID and separate scr and dst ports when processing segments.

dmitrii-beliakov opened the pull request:
This change fixes the bug when redundant ports appeared in the flows by port response.

closes #5450

niksv commented on src-java/kilda-persistence-tinkerpop/src/main/java/org/openkilda/persistence/ferma/repositories/FermaFlowRepository.java:
```diff
@@ -560,11 +560,13 @@ public Map<Integer, Collection<Flow>> findSwitchFlowsByPort(SwitchId switchId, C
-
+            log.error("FermaFlowRepository add end points: {}", portToFlowMap);
             portToFlowMap.entrySet().removeIf(e -> (ports != null && !ports.contains(e.getKey())));
+            log.error("FermaFlowRepository remove not needed ports {}: {}", ports, portToFlowMap);
```
Did you forget to remove debug logs?

dmitrii-beliakov replied:
yes, I'll remove it now

niksv reviewed (commented):
Could you please add unit tests?

dmitrii-beliakov commented:
> Could you please add unit tests?

I am thinking about how to do that. Without DB it doesn't seem useful, some integration test could catch things like this bug, but we need to spin up DB for that.

niksv commented:
@dmitrii-beliakov We have `FermaFlowRepositoryTest` 
As far as I remember in runs in memory DB

dmitrii-beliakov commented:
> @dmitrii-beliakov We have `FermaFlowRepositoryTest` As far as I remember in runs in memory DB

I adjusted a unit test. The test didn't catch the bug, so I first thought that it is the env issue. But it turned out that in reality it was the test data masking this bug. Just unfortunate select of values and corner cases. Now I split the cases into separate tests.

pkazlenka commented:
Verified manually. No need to add functional autotest for this small bugfix.
